Bug: pause watchdog punching in test_watchdog.py for Nexthop devices

### Is it platform specific

generic

### Importance or Severity

High

### Description of the bug

The [pytest fixture](https://github.com/sonic-net/sonic-mgmt/blob/005e4c81cfc8bed6e8c3a6d195569a53300e194b/tests/platform_tests/api/test_watchdog.py#L47) requires the watchdog to be disarmed at the beginning
of each test case. Nexthop devices have watchdog enabled by default and armed by a puncher periodically. 

Therefore, the watchdog puncher should be paused and the watchdog disarmed during each test case. The watchdog puncher should be paused to also avoid race conditions in test cases that asserts conditions based on `arm` and `disarm` calls.

### Steps to Reproduce

- Run test_watchdog.py on a Nexthop device
- `watchdog_not_running` fixture fails due to watchdog being armed
- Even if disarmed, observe race conditions due to watchdog punching arming the watchdog and `disarm` call ([example](https://github.com/sonic-net/sonic-mgmt/blob/005e4c81cfc8bed6e8c3a6d195569a53300e194b/tests/platform_tests/api/test_watchdog.py#L132-L136))

### Actual Behavior and Expected Behavior

- Run test_watchdog.py on a Nexthop device
- All test cases pass with no random race conditions

### Relevant log output

```shell
E           Failed: Failed to arm the watchdog, Watchdog remaining_time 9 seconds appears wrong compared to watchdog timeout -1 seocnds, Failed to disarm the watchdog, Watchdog is not disarmed, Watchdog remaining_time 9 seconds is wrong for disarmed state, unexpected disconnection from dut
```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/platform_tests/api/test_watchdog.py
+++ b/tests/platform_tests/api/test_watchdog.py
@@ -50,9 +50,14 @@ class TestWatchdogApi(PlatformApiTestBase):
         and disables it after the test ends'''
 
         duthost = duthosts[enum_rand_one_per_hwsku_hostname]
-        if duthost.facts['platform'] == 'armhf-nokia_ixs7215_52x-r0' or \
-                duthost.facts['platform'] == 'arm64-nokia_ixs7215_52xb-r0' or \
-                duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_dpu"):
+        if (
+            duthost.facts['platform'] == 'armhf-nokia_ixs7215_52x-r0'
+            or duthost.facts['platform'] == 'arm64-nokia_ixs7215_52xb-r0'
+            or duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_dpu")
+        ):
+            duthost.shell("watchdogutil disarm")
+        elif duthost.facts["platform"].startswith("x86_64-nexthop_"):
+            duthost.shell("systemctl disable watchdog.timer --now")
             duthost.shell("watchdogutil disarm")
 
         assert not watchdog.is_armed(platform_api_conn)
@@ -65,6 +70,9 @@ class TestWatchdogApi(PlatformApiTestBase):
             if duthost.facts['platform'] == 'armhf-nokia_ixs7215_52x-r0' or \
                     duthost.facts['platform'] == 'arm64-nokia_ixs7215_52xb-r0':
                 duthost.shell("systemctl start cpu_wdt.service")
+            elif duthost.facts["platform"].startswith("x86_64-nexthop_"):
+                duthost.shell("systemctl enable watchdog.timer --now")
+
             if duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_dpu"):
                 duthost.shell("watchdogutil arm")
 

